kdumpctl

```diff
@@ -1672,47 +1675,162 @@ _should_reset_crashkernel() {
 	[[ $(kdump_get_conf_val auto_reset_crashkernel) != no ]] && systemctl is-enabled kdump &> /dev/null
 }
 
+set_kdump_test_id()
+{
+	local _id=$1
+	local _kmsg="/dev/kmsg"
+
+	if [[ $DEFAULT_DUMP_MODE == "fadump" ]]; then
+		[[ -c "$_kmsg" ]] && echo "$_id" > "$_kmsg"
+	else
```

Comment:
I noticed that @hbathini just added a mechanism to append parameters to the fadump kernel command line in
77b80ce ("fadump: pass additional parameters for capture kernel"). Using that would make this feature much simpler as well. As then kdump and fadump can be handled the same way.

Especially it would make it trivial to parse the `kdump_test_id` from the command line within the initrd for both kdump and fadump. That's why I believe it would make the whole implementation simpler and less error prone if some of the functionality is moved to the initrd.

What I have in mind is:
1) `kdumpctl test` sets the `kdump_test_id` like in your patch.
2) the initrd parses the `kdump_test_id` from the command line and creates a temporary `$path/.$kdump_test_id.status' file to write the status to. As the initrd knows when dumping has succeeded it could simply initialize the file to 'fail' and update it to 'success' right before calling the final_action.
3) When `kdumpctl` fetches the status it only needs to check whether this file exists and then can simply read the status from it without parsing a whole bunch of `dmesg` files.

The only "problem" I see is that `kdump.sh` probably will need a small refractoring/cleanup for ssh dumps. But that is probably something we should do anyway.

What do you think?

Reply:
Hi @prudo1 ,

Thanks for the comments! I have no objection for your approach itself, however there is only one concerns from my side, the fadump support of additional parameter for 2nd kernel is a new feature, which requires kernel patch (3416c9daa6b13 "powerpc/fadump: pass additional parameters when fadump is active") etc backporting. I'm not sure whether the kernel patch will be backported to rhel9 or rhel8 or some older version of rhel9.1/2/3..., since CEE requires the "kdumpctl test" feature mainly to support older kernels in their daily work, and I guess if CEE think the "kdumpctl test" feature is good enough, they probably would ask us to support it in rhel8 as well.

If they do, this will make a burden for us to maintain both traditional & newer fadump. Dave suggested me to ask if CEE wants "kdumpctl test" for fadump, because comparing to kdump, fadump is more stable and probably don't need "kdumpctl test" at all. Before CEE comfirmation, I would suggest to keep "kdumpctl test" minimal dependency for easier backporting, what do you think?

Reply:
Other than passing the kdump_test_id string through command line to 2nd kernel, I cannot think of a better way to do that. Packing the kdump_test_id string into initrd img as a file is also not a good idea, this will ask initrd rebuild every time for kdumpctl test. The rebuilt initrd will not reflect the true env of what we are trying to test.

Or 2nd kernel doesn't have to know kdump_test_id at all, just save the vmcore as it used to do. After vmcore saved success, it can set a mark as success/failure next to vmcore. When back to 1st kernel, we can fetch for the mark. But this approach have no difference with current approach, vmcore itself is the success/failure mark already...

Reply:
Hi @liutgnu,

you are right, that depending on the kernel patch isn't ideal but I wouldn't worry about it too much. The way I see it powerpc and fadump are only used by a small number of our users. So the question is if it is worth adding so much complexity, which we need to support in the future, only because a small number of users might need it for older releases. Especially as the kernel commit in question has already been backported to c9s. So all future releases will support it. Personally I'm tending more to keep the implementation simple and not support fadump on older releases.
And in case CEE (or whoever) thinks that fadump support is needed even for older releases and get's the exception to include it. Then we still can find a way to get it to work there. I just don't think that we should include it into the 'standard' implementation we'll need to support for the next 10+ years.

True, there's no suitable alternative other than passing it via the command line. That's why I don't think it makes sense to parse all the dmesg files to find the id when there is all the infrastructure in place to parse the id from the command line in the initrd and use it to simplify the way we report the result back to the production system.

Reply:
Hi @prudo1 ,

Thanks for the comments, agreed. In addition, CEE replied to me that they don't need "kdumpctl test" for fadump, so I removed fadump related part in v3, please check it out. Thanks!